### ISSUE TITLE
Meta information lost in i/o of Spectral Datasets

### ISSUE THREAD

**Issue by AtreyeeS** [gammapy/gammapy#2545]
Explicitly defined meta information on the datasets is lost in `to_ogip_files`.
This is a problem, for example, in lightcurve analysis, where we add the `t_start` and `t_stop` info on the counts metadata. 
```
    for time_interval in time_intervals:
        observation = observations.select_time(time_interval)[0]
        dataset = dataset_maker.run(observation, selection=["counts", "aeff", "edisp"])
        dataset.counts.meta = dict()
        dataset.counts.meta["t_start"] = time_interval[0]
        dataset.counts.meta["t_stop"] = time_interval[1]
```
This issue is linked to #2542 and will probably be addressed together, but creating as a new one since this is a serious impediment to lc analysis.

**Comment by cdeil** [comment]
@registerrier said he'd take care of LC - assigning to you.

My 2 cents: putting the burden un users to script time processing themselves in data reduction and touch any `meta` dict is really annoying and error-prone. The simple solution is to script this in Gammapy (see 
 https://github.com/gammapy/gammapy/pull/1451#issuecomment-546274654).

But I don't want to argue again. Unsubscribing now.

**Comment by adonath** [comment]
If I remember correctly adding the  `t_start` and `t_stop` to `.meta` was just a preliminary "hack" to get the light curve estimation working at all. By now I think we can simply change to use `dataset.gti` in the `LightCurveEstimator` instead? Is this correct @registerrier?

**Comment by AtreyeeS** [comment]
Adapting `LightCurveEstimator` would indeed be the best solution because I get a similar problem while writing a `MapDataset` with time info on the meta as well:
`TypeError: Object of type Time is not JSON serializable`

But, since we are at it, 1D spectra meta info should not get lost during serialisation, right?

**Comment by registerrier** [comment]
Yes exactly, the `LightCurveEstimator` must be changed to get the time directly from the `GTI` data that is now properly stored on the `Dataset`. The `GTI` table is serialized as a separate extension, even with the `to_ogip_files` methods.

**Comment by registerrier** [comment]
The issue of the time serialization is solved now with #2561 . 

Keeping the issue open for now. Need to decide what to do with `meta` stored on `dataset.counts`

**Comment by registerrier** [comment]
`SpectrumDatasetOnOff.to_ogip_file()` has a specific export for `meta` data to make the files compliant with the OGIP conventions. I suggest not to change this for now.
Closing issue.
